### ISSUE TITLE
tabyl fails on large dataset: Error: C stack usage  12539755 is too close to the limit

### ISSUE THREAD

**Issue by randomgambit** [sfirke/janitor#190]
Hi,

I am loading a dataframe with about 8 million obs. Running the following query gives

```
> df_flash %>% filter(str_detect(u, 'hello.com')) %>% tabyl(u, sort = TRUE)
Error: C stack usage  12539755 is too close to the limit
```
I have plenty of memory. Is this a tabyl limitation?

Thanks!

**Comment by sfirke** [comment]
Hm. I have not tried tabyl on data frames of that size. If you try
filtering it to be smaller, at what point do you not get that error?

Also the sort = TRUE argument only works on janitor 0.3.1 and earlier,
consider trying this on the version 1.0 now on Github.

**Comment by almartin82** [comment]
That looks like a memory error I have seen when working with dplyr - unless
I am most taken there are no Rcpp functions in janitor, so my guess is that
you are hitting the limit on what your machine can do in the tidyverse in
memory.

**Comment by randomgambit** [comment]
are you kidding my friend? after the filter there are about 80k obs left, very small but the data is likely very sparse as well. I have more than 100GB ram so there is no way i am hitting a machine limitation. I will check shortly which version I have. thanks fellas 😊

**Comment by sfirke** [comment]
Yep under the hood in tabyl is dplyr and tidyr, no C in this package except
what other packages have. So agreed with Andrew that this message is likely
coming from a dplyr or tidyr C call.

**Comment by sfirke** [comment]
I wonder if tracing the error would shed some light on what is occurring
that throws this message

**Comment by randomgambit** [comment]
how can i do that?

**Comment by randomgambit** [comment]
this is kind of ridiculous because running

` df_flash %>% filter(str_detect(u, 'hello.com')) %>% group_by(u) %>% summarize(mycount = n()) %>% ungroup() %>% arrange(desc(mycount))`

works in less than 3 seconds...

**Comment by sfirke** [comment]
Will close this b/c it seems like an issue with underlying C code in dplyr & tidyr.  If you can create a reproducible example with simulated data where we can see janitor::tabyl() failing when dplyr/tidy does not, then that'll be something to debug and we can reopen this.
